### ISSUE TITLE
ResolveHooker class

### ISSUE THREAD

**Issue by jbfp** [fsprojects/fsharp-dnx#16]
I'm trying to get an understanding of the codebase, and I've come across the ResolveHooker class. There's a comment saying it's an ugly hack that should be removed, and I tried commenting it out at https://github.com/fsprojects/fsharp-dnx/blob/master/src/FSharp.Dnx/FSharpCompiler.cs#L66 and it didn't seem to make a difference. Is it still required, do you think, or should it be removed? :)

**Comment by Alxandr** [comment]
Problem is, it's was only ever needed in certain cases. And I don't
remember which ones :P. As far as I know, it's still needed, but let me
look into it and get back to you.

**Comment by jbfp** [comment]
Yeah I figured it was needed for some exotic edge case, but I always jump at a chance patch some "hacks" ;)

**Comment by Alxandr** [comment]
Yeah. Problem is, FSharp.Core is not a dependency of this project, but it's
always in the dependency chain somewhere. So yeah.

**Comment by jbfp** [comment]
But wouldn't it get pulled in when FSharp.Dnx is referenced, since it references FSharp.Core?

**Comment by Alxandr** [comment]
Hmm. You might be right. I think maybe this was needed because `FSharp.Core` used to be referenced as a build-time-only reference...
